src/components/ui/DynamicForm.jsx

```diff
@@ -0,0 +1,175 @@
+import React from 'react'
+import { Input } from '@/components/ui/input'
+import { Label } from '@/components/ui/label'
+import {
+  Select,
+  SelectItem,
+  SelectContent,
+  SelectTrigger,
+  SelectValue,
+} from '@/components/ui/select'
+import { Button } from '@/components/ui/button'
+import {
+  Accordion,
+  AccordionItem,
+  AccordionTrigger,
+  AccordionContent,
+} from '@/components/ui/accordion'
+
+const DynamicForm = ({ fields, formData, errors, onChange, mode }) => {
+  const renderField = field => {
+    const { name, label, type, placeHolder, options = [] } = field
+    const value = formData[name] || ''
+
+    const commonClasses = 'space-y-1 w-full'
+
+    switch (type) {
+      case 'text':
+      case 'textarea':
+        return (
+          <div key={name} className={commonClasses}>
+            <Label htmlFor={name}>{label}</Label>
+            {type === 'textarea' ? (
```

Comment:
Instead of putting an if-else check here, can't we write separate definitions for text and textarea?? It makes the code more readable